src/clj/runbld/main.clj

```diff
@@ -41,6 +42,28 @@
      ;; for tests when #'really-die is redefed
      msg*)))
 
+(defn bootstrap-workspace [raw-opts]
+  (let [clone? (boolean (-> raw-opts :scm :clone))
+        wipe-workspace? (boolean (-> raw-opts :scm :wipe-workspace))
+        local (-> raw-opts :process :cwd)
+        remote (-> raw-opts :scm :url)
+        reference (-> raw-opts :scm :reference-repo)
+        branch (-> raw-opts :scm :branch)
+        depth (-> raw-opts :scm :depth)]
+    (when clone?
+      (let [clone-args (->> [(when reference ["--reference" reference])
+                             (when branch ["--branch" branch])
+                             (when depth ["--depth" (str depth)])]
+                            (filter identity)
+                            (apply concat))]
+        (when wipe-workspace?
+          (let [workspace (System/getenv "WORKSPACE")]
+            (io/log "wiping workspace" workspace)
+            (clojure.java.shell/sh "find" workspace "-mindepth" "1" "-delete")))
```

Comment:
Hm, just occurred to me that I have no idea if this will work on Windows. I'll look into doing it in a more Javaesque way.